### PR TITLE
fix: signin and signup fixes. add duplicate record exception

### PR DESCRIPTION
**Pull request by capan**

### CODE REVIEW

File: src/services/auth.service.ts [hunk 1]
```diff
@@ -17,7 +17,7 @@ export interface Token {
 export default class AuthService extends BaseService {
   knx: KnexDB;
   userRepository: UserRepository;
-  tokens: Token;
+  tokens: Record<string, unknown>;
```
Comment (zhunor) [inline src/services/auth.service.ts:20]: Why we give up on using `Token` interface defined above, i think it is better to have explicit type for this one
https://github.com/capan/mapptastic-be/pull/9/files#diff-40f7172e540c3b96f51825110cded9d3c48f0b5b3726aaa6938b072fcd430225R12

File: src/services/auth.service.ts [hunk 2]
```diff
@@ -40,17 +40,30 @@ export default class AuthService extends BaseService {
...
+          .insert(user);
+        this.tokens.accessToken = this.userRepository.accessTokenForUser(user);
+        this.tokens.refreshToken = this.userRepository.refreshTokenForUser(user);
+        result = this.response.Success('Sign up successfull.', {
+          accessToken: this.userRepository.accessTokenForUser(user),
+          refreshToken: this.userRepository.refreshTokenForUser(user)
+        });
+        return result
       } catch (err) {
         console.error(`Signup err ${err}`);
```
Comment (zhunor) [inline src/services/auth.service.ts:62]: I don't necessarily think that you wanna `console.error` here as you are already throwing an exception below. I think this is one my temporary additions. You can delete it

**Review by zhunor — CHANGES_REQUESTED** [review]

File: src/services/auth.service.ts (hunk 1 above)
Comment (capan) [reply src/services/auth.service.ts]: Token type was mistakenly removed. Now that mistake reverted

File: src/services/auth.service.ts (hunk 2 above)
Comment (capan) [reply src/services/auth.service.ts]: Express error handler logs the trowed error. `console.error` is surely unnecessary 👍🏼. removed

File: src/services/auth.service.ts [hunk 3]
```diff
@@ -40,17 +40,29 @@ export default class AuthService extends BaseService {
     return this.response.Exception('Access token required');
   }
 
-  async signup(user: IUser): Promise<void> {
+  async signup(user: IUser): Promise<IResponse | IError> {
     const userCheck = await this.userRepository.userCheck(user.email);
+    const saltRound = 10;
+    let result;
     if (!userCheck) {
+      // const hashedPassword = await bcrypt.hash(password, saltRound);
```
Comment (zhunor) [inline src/services/auth.service.ts:48]: We can get rid of this comment, same below

**Review by zhunor — CHANGES_REQUESTED** [review]

File: src/services/auth.service.ts (hunk 3 above)
Comment (capan) [reply src/services/auth.service.ts:48]: done!

Comment (capan) [reply src/services/auth.service.ts]: done

**Review by zhunor — APPROVED** [review]
LGTM ✅

**Pull request merged**